packages/next-swc/crates/next-core/src/next_font/local/request.rs

```diff
@@ -0,0 +1,101 @@
+use serde::{Deserialize, Serialize};
+use turbo_tasks::trace::TraceRawVcs;
+
+/// The top-most structure encoded into the query param in requests to
+/// `next/font/local` generated by the next/font swc transform. e.g.
+/// `next/font/local/target.css?{"path": "index.js", "arguments": {"src":...
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(super) struct NextFontLocalRequest {
+    pub arguments: (NextFontLocalRequestArguments,),
+    pub variable_name: String,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(super) struct NextFontLocalRequestArguments {
+    pub src: SrcRequest,
+    pub weight: Option<String>,
+    pub style: Option<String>,
+    #[serde(default = "default_display")]
+    pub display: String,
+    #[serde(default = "default_preload")]
+    pub preload: bool,
+    pub fallback: Option<Vec<String>>,
+    #[serde(
+        default = "default_adjust_font_fallback",
+        deserialize_with = "deserialize_adjust_font_fallback"
+    )]
+    pub adjust_font_fallback: AdjustFontFallback,
+    pub variable: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+pub(super) enum SrcRequest {
+    One(String),
+    Many(Vec<SrcDescriptor>),
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub(super) struct SrcDescriptor {
+    pub path: String,
+    pub weight: Option<String>,
+    pub style: Option<String>,
+}
+
+#[derive(
+    Clone, Debug, Deserialize, Hash, Ord, PartialOrd, PartialEq, Eq, Serialize, TraceRawVcs,
+)]
+pub(super) enum AdjustFontFallback {
+    Arial,
+    TimesNewRoman,
+    None,
+}
+
+fn default_adjust_font_fallback() -> AdjustFontFallback {
+    AdjustFontFallback::Arial
+}
+
+fn deserialize_adjust_font_fallback<'de, D>(
```

Comment:
Can you write a unit test for this deserializer? Something like https://github.com/vercel/turbo/blob/5b32049998ad491566154c037ce0d2d4b9273965/crates/turbo-tasks-bytes/src/bytes.rs#L94-L95